storage: rework TestProposalOverhead

tbg opened the pull request:
It flaked on me (as I suppose it sometimes does), so the next time
I'll want to see something actionable. The printout of the command
should achieve that because we can compare it to that of a passing
run.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/36653)
<!-- Reviewable:end -->

petermattis commented on pkg/storage/replica_test.go:
```diff
@@ -6265,17 +6265,26 @@ func TestBatchErrorWithIndex(t *testing.T) {
+				return nil
+			}
+			atomic.StoreUint32(&overhead, uint32(args.Cmd.Size()-args.Cmd.WriteBatch.Size()))
+			args.Cmd.WriteBatch = nil // shallow modification
```
What is this about? This addition seems more significant than the commit message indicates.

tbg replied:
I don't want to `pretty.Sprint` the `WriteBatch` because it's intentionally excluded from the size computation. `args` is a value, and `Cmd` isn't a pointer neither, so we're not changing the data the caller sees, which I was trying to indicate in the comment. Does that make sense?

petermattis replied:
Ah, got it. The comment wasn't helping me there. Perhaps `// avoid printing the WriteBatch (shallow modification)`.

tbg replied:
I went full verbosity.

tbg committed:
storage: rework TestProposalOverhead

It flaked on me (as I suppose it sometimes does), so the next time
I'll want to see something actionable. The printout of the command
should achieve that because we can compare it to that of a passing
run.

Release note: None

tbg commented:
TFTR!

bors r=petermattis
